agent/01_agent_install_requirements.sh

```diff
@@ -0,0 +1,28 @@
+# #!/usr/bin/env bash
+set -euxo pipefail
+
+XDG_CONFIG_HOME=${XDG_CONFIG_HOME:-$(systemd-path user-configuration)}
+SCRIPTDIR="$(pwd)"
+USER=`whoami`
+
+# Get variables from the config file
+if [ -z "${CONFIG:-}" ]; then
+    # See if there's a config_$USER.sh in the SCRIPTDIR
+    if [ -f ${SCRIPTDIR}/config_${USER}.sh ]; then
+        echo "Using CONFIG ${SCRIPTDIR}/config_${USER}.sh" 1>&2
+        CONFIG="${SCRIPTDIR}/config_${USER}.sh"
+    elif [[ -f "${XDG_CONFIG_HOME}/dev-scripts/config" ]]; then
+        echo "Using CONFIG ${XDG_CONFIG_HOME}/dev-scripts/config" 1>&2
+        CONFIG="${XDG_CONFIG_HOME}/dev-scripts/config"
+    else
+        error "Please run with a configuration environment set."
+        error "eg CONFIG=config_example.sh ./01_all_in_one.sh"
+        exit 1
+    fi
+fi
+source $CONFIG
+
```

Comment:
I suggest having a more informative error when AGENT_CONFIG is not set. This is what I'm seeing atm:
```
./agent/01_agent_install_requirements.sh: line 25: AGENT_CONFIG: unbound variable
make: *** [Makefile:30: agent_requirements] Error 1
```